credoai/artifacts/model/classification_model.py

```diff
@@ -81,42 +88,50 @@ def __post_init__(self):
         elif self.model_info["framework"] in MLP_FRAMEWORKS:
             # TODO change this to '__call__' when adding in general TF and PyTorch
             pred_func = getattr(self, "predict", None)
-            if pred_func:
-                if self.model_like.layers[-1].output_shape == (None, 1):
-                    self.__dict__["predict"] = pred_func
-                else:
-                    self.__dict__["predict"] = lambda x: np.argmax(pred_func(x), axis=1)
-
-                if self.model_like.layers[-1].output_shape == (None, 2):
-                    self.__dict__["predict_proba"] = lambda x: pred_func(x)[:, 1]
-                elif (
-                    len(self.model_like.layers[-1].output_shape) == 2
-                    and self.model_like.layers[-1].output_shape[1] > 2
-                ):
-                    self.__dict__["predict_proba"] = pred_func
-                else:
-                    pass
-                    # predict_proba is not valid (for now)
+            if self.model_like.layers[-1].output_shape == (None, 1):
```

Comment:
Why was the check for `pred_func` not needed?

Reply:
It is. Added back.